examples/benchmark/main.go

```diff
@@ -0,0 +1,202 @@
+		if err != nil {
+			return fmt.Errorf("error starting cpu profile: %w", err)
+		}
+		defer pprof.StopCPUProfile()
+	}
+
+	accountKey := keypair.MustRandom()
+	escrowAccountKey := keypair.MustRandom()
+
+	horizonClient := &horizonclient.Client{HorizonURL: horizonURL}
+	networkDetails, err := horizonClient.Root()
+	if err != nil {
+		return err
+	}
+
+	streamer := &horizon.Streamer{
+		HorizonClient: horizonClient,
+		ErrorHandler: func(err error) {
+			fmt.Fprintf(os.Stderr, "horizon streamer error: %v\n", err)
+		},
+	}
+	balanceCollector := &horizon.BalanceCollector{HorizonClient: horizonClient}
+	sequenceNumberCollector := &horizon.SequenceNumberCollector{HorizonClient: horizonClient}
+	submitter := &submit.Submitter{
+		SubmitTxer:        &horizon.Submitter{HorizonClient: horizonClient},
+		NetworkPassphrase: networkDetails.NetworkPassphrase,
+		BaseFee:           txnbuild.MinBaseFee,
+		FeeAccount:        accountKey.FromAddress(),
+		FeeAccountSigners: []*keypair.Full{accountKey},
+	}
+
+	fmt.Fprintf(os.Stdout, "creating main account %s with network root\n", accountKey.Address())
+	err = createAccountWithRoot(horizonClient, networkDetails.NetworkPassphrase, accountKey)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(os.Stdout, "creating escrow account %s with network root\n", escrowAccountKey.Address())
+	err = createAccountWithSignerWithRoot(horizonClient, networkDetails.NetworkPassphrase, escrowAccountKey, accountKey.FromAddress())
+	if err != nil {
+		return err
+	}
+
+	// Wait for state of escrow accounts to be ingested by Horizon.
+	time.Sleep(2 * time.Second)
+
+	events := make(chan agentpkg.Event)
+	config := agentpkg.Config{
+		ObservationPeriodTime:      observationPeriodTime,
+		ObservationPeriodLedgerGap: observationPeriodLedgerGap,
+		MaxOpenExpiry:              maxOpenExpiry,
+		NetworkPassphrase:          networkDetails.NetworkPassphrase,
+		SequenceNumberCollector:    sequenceNumberCollector,
+		BalanceCollector:           balanceCollector,
+		Submitter:                  submitter,
+		Streamer:                   streamer,
+		EscrowAccountKey:           escrowAccountKey.FromAddress(),
+		EscrowAccountSigner:        accountKey,
+		LogWriter:                  io.Discard,
+		Events:                     events,
+	}
+	agent := agentpkg.NewAgent(config)
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		paymentsSent := 0
+		paymentsReceived := 0
+		var timeStart time.Time
+		for {
+			switch e := (<-events).(type) {
+			case agentpkg.ErrorEvent:
+				fmt.Fprintf(os.Stderr, "%v\n", e.Err)
+			case agentpkg.ConnectedEvent:
+				fmt.Fprintf(os.Stderr, "connected\n")
+				if connectAddr != "" {
+					_ = agent.Open()
+				}
+			case agentpkg.OpenedEvent:
+				fmt.Fprintf(os.Stderr, "opened\n")
+				if connectAddr != "" {
+					tick := time.Tick(1 * time.Second)
+					for i := 5; i >= 0; {
+						fmt.Fprintf(os.Stderr, "%d\n", i)
+						i--
+						if i >= 0 {
+							<-tick
+						}
+					}
```

Comment:
❓ a little confused by this part. Is this just counting down from 5?

Reply:
It's just a countdown at the beginning to make sure both ends are setup. There's an issue where one side may not be ready and have ingested the open and updated their internal state. Rather than add code to handle that more gracefully I added this delay.